### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by sjoerd0301**
Update android location permission
remove kotlin libs
cancel locationstatus listening on dispose
update ui in regards to the painted marker, also reflect options.showHeading in the marker
delete useless test

### CODE REVIEW

File: lib/src/user_location_layer.dart [hunk 1]
```diff
@@ -66,14 +68,10 @@ class _MapsPluginLayerState extends State<MapsPluginLayer>
...
-          if (_onLocationChangedStreamSubscription != null) {
-            _onLocationChangedStreamSubscription.cancel();
-          }
+          _cancel(_onLocationChangedStreamSubscription);
           break;
         case AppLifecycleState.resumed:
-          if (_onLocationChangedStreamSubscription != null) {
-            _onLocationChangedStreamSubscription.resume();
-          }
+          _cancel(_onLocationChangedStreamSubscription);
```
Comment (igaurab) [inline lib/src/user_location_layer.dart:74]: Although this works, why are we closing the stream in both the cases? Just curious

**Review by igaurab — APPROVED** [review]

File: lib/src/user_location_layer.dart (hunk 1 above)
Comment (sjoerd0301) [reply lib/src/user_location_layer.dart:74]: Oh, sorry I must have missed that, my bad

Comment (igaurab) [reply lib/src/user_location_layer.dart:74]: I am sorry for the trouble, but could you send a PR to fix it,  if possible?

Comment (sjoerd0301) [reply lib/src/user_location_layer.dart:74]: No problem, I don't have access to my pc right now, but I'll see if I can do it tomorrow

Comment (igaurab) [reply lib/src/user_location_layer.dart:74]: Thanks, my laptop is kinda broke so...running OS from the live-image as of now

**Pull request merged**